File: src/app/services/instruments.service.ts
import { Injectable, signal } from '@angular/core'
import { Instrument } from '../models/instrument'

@Injectable({
  providedIn: 'root',
})

export class InstrumentsService {
  instruments = signal<Instrument[]>([
    { id: 1, label: 'Piano / Órgano', path: 'piano', name: 'piano'},
    { id: 2, label: 'Violin 1', path: 'violin1', name: 'violin'  },
    { id: 3, label: 'Violin 2', path: 'violin2', name: 'violin'  },
    { id: 4, label: 'Flauta 1', path: 'flute1',  name: 'flute' },
    { id: 5, label: 'Flauta 2', path: 'flute2',  name: 'flute' },
    { id: 6, label: 'Trompeta', path: 'trumpet',  name: 'trumpet' },
    { id: 8, label: 'Clarinete 1', path: 'clarinet1',  name: 'clarinet' },
    { id: 9, label: 'Clarinete 2', path: 'clarinet2',  name: 'clarinet' },
    { id: 10, label: 'Viola', path: 'viola', name: 'viola' },
    { id: 11, label: 'Cello', path: 'cello', name: 'cello' },
  ])

  getInstrument(id: number): Instrument {
    return this.instruments().find((instrument) => instrument.id === id)
  }
}

File: new_symphony/lib/core/services/instruments_service.dart
import 'package:new_symphony/data/models/instrument.dart';

class InstrumentsService {
  final List<Instrument> _instruments = [
    Instrument(id: 1, name: 'Piano', path: 'piano', iconPath: 'assets/icons/piano.svg'),
    Instrument(id: 2, name: 'Violín 1', path: 'violin1', iconPath: 'assets/icons/violin.svg'),
    Instrument(id: 3, name: 'Violín 2', path: 'violin2', iconPath: 'assets/icons/violin.svg'),
    Instrument(id: 4, name: 'Trompeta', path: 'trumpet', iconPath: 'assets/icons/trumpet.svg'),
    Instrument(id: 5, name: 'Flauta 1', path: 'flute1', iconPath: 'assets/icons/flute.svg'),
    Instrument(id: 6, name: 'Flauta 2', path: 'flute2', iconPath: 'assets/icons/flute.svg'),
    Instrument(id: 7, name: 'Clarinete 1', path: 'clarinet1', iconPath: 'assets/icons/clarinet.svg'),
    Instrument(id: 8, name: 'Clarinete 2', path: 'clarinet2', iconPath: 'assets/icons/clarinet.svg'),
    Instrument(id: 9, name: 'Viola', path: 'viola', iconPath: 'assets/icons/viola.svg'),
    Instrument(id: 10, name: 'Cello', path: 'cello', iconPath: 'assets/icons/cello.svg'),
  ];

  List<Instrument> instruments() => _instruments;

  Instrument getInstrument(int id) => 
      _instruments.firstWhere((i) => i.id == id, orElse: () => _instruments.first);
}